### ISSUE TITLE
TEXTURE is always white on some control nodes

### ISSUE THREAD

**Issue by Ev01** [godotengine/godot#76312]
### Godot version

4.0.2

### System information

Windows 10, NVIDIA GeForce GT 640M, Compatibility

### Issue description

As far as I know, the following shader should output the original sprite/picture.
```
void fragment() {
	COLOR = texture(TEXTURE, UV);
}
```
Putting this on a Sprite node or TextureRect doesn't change the image, but putting it on other Control nodes such as Panels and ColorRects turns them white.

Expected behaviour:
![image](https://user-images.githubusercontent.com/42711659/233576999-daed6f90-dc3a-428d-8e2e-690dd0ac0713.png)

Actual behaviour:
![image](https://user-images.githubusercontent.com/42711659/233577305-fde93be4-ac6d-4dde-a6bb-b4757ba53ac7.png)

### Steps to reproduce

1. Download minimal reproduction project.
2. Double click "nothing_material.tres" and drag "nothing.gdshader" onto the shader variable in the inspector. The Panel, ColorRect, Sprite, and TextureRect all use this material
3. The Sprite and TextureRect don't change, but the panel and ColorRect both turn white.

### Minimal reproduction project

[ShaderTest.zip](https://github.com/godotengine/godot/files/11293641/ShaderTest.zip)

**Comment by KoBeWi** [comment]
See #34368
Only nodes with actual texture use the TEXTURE uniform.

**Comment by Arecher** [comment]
Apologies for raising this one from the dead, but it seems to be the most official location that discusses this limitation, apart from the linked thread, but that one seems more outdated at this point.

I ran into this limitation as well, and while I completely understand that `Control` nodes without a texture cannot be expected to return their color via `TEXTURE` in shaders, that leaves the question: **How are you meant to get the displayed color of a Control node?**

Since you are allowed to put Shaders on to Control nodes, which can be very useful, how are you meant to write a shader that can handle both textured Control nodes and untextured ones such as `PanelContainers` with a `StyleBoxFlat`. It's extremely confusing that there is no alternative way to get the color that is displayed.

Obviously it would be possible to use a `uniform` and pass the value, whether that's a single color or a Viewport capture image. But that seems extremely convoluted and shouldn't be necessary, since for any textured Control node a built-in function exist with `texture(TEXTURE, UV)`.

The only option I have found is to use:
```
float alphavalue = 0.5; // A hardcoded alpha value, or any calculation such as a lerp
COLOR.a = alphavalue;
```

This way the shader does not overwrite the `.rgb` values of the `COLOR` and the original color is preserved. But this clearly doesn't work for all shading cases. I can write a proposal for the addition of a method to get the equivalent of `texture(TEXTURE, UV)`, but I thought I would ask around first, because there MUST be a way to get these color values that I've completely overlooked.
